pyteal/ast/opup.py

```diff
@@ -0,0 +1,141 @@
+from pyteal.ast.app import OnComplete
+from .while_ import While
+from .expr import Expr
+from .global_ import Global
+from .seq import Seq
+from .if_ import If
+from .int import Int
+from .bytes import Bytes
+from .gitxn import Gitxn
+from .itxn import InnerTxnBuilder
+from .scratch import ScratchSlot
+from .txn import TxnField, TxnType, Txn
+from enum import Enum
+
+
+class OpUpMode(Enum):
+    """An Enum object that defines the mode used for the OpUp utility.
+       
+       Note: the Explicit and OnCreate modes require the app id to be provided
+       through the foreign apps array.
+    """
+
+    """The app to call must be provided by the user."""
+    Explicit = 0
+
+    """The app to call is created/deleted when the main app is created/deleted."""
+    OnCreate = 1
+
+    """The app to call is created then deleted for each request to increase budget."""
+    OnCall = 2
+
+
+NoOpApp = Bytes("byte 0x068101") # teal program "int 1" assembled
```

Comment:
If this is used as the app approval program for `OnCreate`, then _anyone_ could update or delete this app, since it does not check who the sender is.

Secondary concern, why is `byte` part of the program string?

Reply:
This was more of a placeholder than anything so haven't put much thought into it. Point taken though.

Reply:
fixed. resolving.